src/posit/connect/users.py

```diff
@@ -25,7 +27,7 @@ class User(TypedDict, total=False):
     locked: bool
 
 
-class Users:
+class Users(Resources[User]):
```

Comment:
What use is it to have `Resources[User]` vs. a simple `Resources` base class? Can/should you use the `T` type in the method definitions? Do you need to define the return types for the methods, or are they redundant to the base class?

More meta question: supposing you could factor some of those things out of the method definitions, should you? I could see an advantage of the current version being that it is explicit (famously better than implicit in Python ;), that you can read the class definition and see what it's doing.

Reply:
I'm not sure I follow. There may be something I need to learn about generics in Python.

Using a parametrized type in the ABC enforces that the inheriting class has consistent return types across methods. Are you suggesting that there are scenarios where this won't be the case?

Reply:
I guess one place I'm confused is that we're still defining the return types in this subclass (e.g. L38, find() returns `List[User]`). Can/should those type annotations be removed because the base class will handle them for you?

My other question is: for example, on L40, we call `User(**user)`, can/should that be `T(**user)`?

The other way I've seen this done is that you would set `self._resource_type = User` or something and use that to construct the objects. (Not saying that's worth doing necessarily.)

Reply:
In this case, consider `Resources` an interface instead of an abstract class since it only has abstract methods. If `Resources` contained non-abstract methods, then `T(**user)` should be the correct usage for conversion in those methods (although it would be `T(**result)`). 

I believe `self._resource_type = User` pattern was correct before generics were introduced in Python. This would have been the only way to inform the base class of the implementing type.

> Can/should those type annotations be removed because the base class will handle them for you?

They can, but when writing the implementing code, the Python language server will auto-complete with the correct type definitions.